enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/BaseHighLimitRecordFormat.java

```diff
@@ -105,10 +103,29 @@ protected final void doRead( RECORD record, PageCursor primaryCursor, int record
                 // Return and try again
                 return;
             }
-            else
+
+            int primaryEndOffset = calculatePrimaryCursorEndOffset( primaryCursor, recordSize );
+
+            // This is a record that is split into multiple record units. We need a bit more clever
+            // data structures here. For the time being this means instantiating one object,
+            // but the trade-off is a great reduction in complexity.
+            long secondaryId = Reference.decode( primaryCursor, PAGE_CURSOR_ADAPTER );
+            try ( SecondaryPageCursorReadDataAdapter readAdapter =
+                    new SecondaryPageCursorReadDataAdapter( primaryCursor, storeFile,
+                            pageIdForRecord( secondaryId, storeFile.pageSize(), recordSize ),
```

Comment:
Can we please extract `#pageIdForRecord()` and following `#offsetForId()` calls into variables? Otherwise it is quite hard to read the code.